### ISSUE TITLE
Automatic Agent Updater (teleport-ent-updater) Downgrades Agent Version

### ISSUE THREAD

**Issue by Clinton-Groundspeed** [gravitational/teleport#27693]
Expected behavior:
The `teleport-ent-updater` tool should only **upgrade** `teleport-ent` (yum/apt package) to a _newer_ version during the maintenance window if a newer version is available.

Current behavior:
The `teleport-ent-updater` tool currently **downgrades** `teleport-ent` (yum/apt package) to an _older_ version during the maintenance window, at least in my Teleport Enterprise Cloud tenant. Servers that were running v13.1.0 were downgraded to v13.0.3 during our Cloud maintenance window, which does not match the tenant version (13.1.0).

Bug details:
- Teleport version: 13.1.0 Enterprise (Cloud)
- Recreation steps: Execute `sudo teleport-upgrade` command on a Linux server (AWS EC2 instance)
- Debug logs:
```
[ec2-user@instance ~]$ sudo teleport-upgrade dry-run
[i] an upgrade is available (13.1.0 -> 13.0.3) [ 392 ]
[i] upgrade is non-critical and we are outside of maintenance window, not attempting. [ 414 ]
```

**Comment by r0mant** [comment]
@Clinton-Groundspeed As we discussed in Slack, the downgrade is the expected behavior if the agent is running version that's newer than the version advertised by the Cloud version lock endpoint:

```
➜  ~ curl https://updates.releases.teleport.dev/v1/stable/cloud/version
v13.0.3
```

This is to support automatic downgrade of agents in case of a bad release. The version will be promoted to 13.1.0 soon, after which your agents should upgrade to it automatically.

Hope this helps and thank you for raising this!

**Comment by fspmarshall** [comment]
`an upgrade is available (13.1.0 -> 13.0.3)` is misleading, downgrading _is_ intended behavior in this case.  That being said, this seems like a reasonable thing to support a customization option for.  Permitting downgrades will continue to be the default option, but it seems reasonable that some folks may want to exceed cloud's current target agent version deliberately while leaving upgrades enabled.
